mountpoint-s3/src/fstab.rs

```diff
@@ -64,36 +68,55 @@ impl FsTabCliArgs {
         }
     }
 
-    fn validate_options(options: &[String]) -> Result<(), clap::Error> {
+    fn validate_options(options: &[String], prefix_allowed: bool) -> Result<(), clap::Error> {
         if options.iter().any(|x| x.starts_with("-")) {
-            return Err(Command::error(
-                &mut Self::command(),
+            return Self::command_error(
                 ErrorKind::InvalidValue,
                 "Cannot prefix arguments with `--` when using fstab style arguments.",
-            ));
+            );
         }
         if options.iter().any(|x| x == "ro") && options.iter().any(|x| x == "rw") {
-            return Err(Command::error(
-                &mut Self::command(),
+            return Self::command_error(
                 ErrorKind::ArgumentConflict,
                 "Cannot use 'ro' flag combined with 'rw' flag.",
-            ));
+            );
         }
-        if let Some(disallowed_option) = options
-            .iter()
-            .find(|x| ["foreground", "read-only"].contains(&x.as_str()))
-        {
-            return Err(Command::error(
-                &mut Self::command(),
+        if options.iter().any(|x| x == "foreground") {
+            return Self::command_error(
                 ErrorKind::InvalidValue,
-                format!(
-                    "Cannot use '{}' flag when using fstab style arguments.",
-                    disallowed_option
-                ),
-            ));
+                "Cannot use 'foreground' flag when using fstab style arguments.",
+            );
+        }
+        if options.iter().any(|x| x == "read-only") {
+            return Self::command_error(
+                ErrorKind::InvalidValue,
+                "Cannot use 'read-only' flag when using fstab style arguments. Use 'ro' instead",
+            );
+        }
+        if !prefix_allowed && options.iter().any(|x| x.starts_with("prefix")) {
+            return Self::command_error(
+                ErrorKind::ArgumentConflict,
+                "Cannot use 'prefix' flag when using s3:// URIs with a prefix",
+            );
         }
         Ok(())
     }
+
+    fn command_error(error_kind: ErrorKind, message: impl fmt::Display) -> Result<(), clap::Error> {
+        Err(Command::error(&mut Self::command(), error_kind, message))
+    }
+
+    fn parse_bucket_prefix(&self) -> (&str, Option<&str>) {
```

Comment:
What about [access point `arn`s](https://github.com/awslabs/mountpoint-s3/blob/main/doc/CONFIGURATION.md#access-points)? Do we need to detect them? Do we want to allow the `s3://` notation?

Reply:
I don't think we should allow the following:
```
s3://arn:aws:s3:::demo_s3_bucket
arn:aws:s3:::demo_s3_bucket/prefix
s3://arn:aws:s3:::demo_s3_bucket/prefix
```
And we should allow `arn:aws:s3:::demo_s3_bucket`. I'll add tests for these